### ISSUE TITLE
ui5 build (without -a) is building all dependencies by default

### ISSUE THREAD

**Issue by sammok83** [SAP/ui5-tooling#798]
## Expected Behavior
When using ui5 build, it shouldn't build all dependencies in the framework section when I'm not specifying -a.

## Current Behavior
When using ui5 build, it shouldn't build all dependencies in the framework section when I'm not specifying -a. However it's doing that in the latest version even without specifying -a

In ui5.yaml:
![image](https://user-images.githubusercontent.com/10509241/223592346-2b489081-38d5-4245-b3c1-997f22c58fbb.png)

In the terminal output:
![image](https://user-images.githubusercontent.com/10509241/223592258-486f40c7-bce3-48e2-a1b8-5e7b17db2ca1.png)

## Steps to Reproduce the Issue
1. Execute **ui5 build preload --clean-dest --include-task=generateCachebusterInfo** / **ui5 build --clean-dest --include-task=generateCachebusterInfo**

## Context
 * UI5 Module Version (output of `ui5 --version` when using the CLI): **`3.0.3`**
 * Node.js Version: **`v18.12.1`**
 * npm Version: `9.6.0`
 * OS/Platform: Windows 10
 * Browser *(if relevant)*:  Chrome

## Log Output / Stack Trace
![image](https://user-images.githubusercontent.com/10509241/223592546-95f5286a-5b48-47a3-a270-10914917886f.png)

**Comment by flovogt** [comment]
Hi @sammok83,
thanks for opening this issue with us. Could you share the complete `ui5.yaml` and `package.json` with us, please?
With UI5 Tooling 3.0 custom tasks has to declare if they need dependencies or not. So I guess, the project is using one which is not adjusted yet. See also here https://github.com/SAP/ui5-tooling/issues/781#issuecomment-1431254954.

**Comment by sammok83** [comment]
Sure:

**ui5.yaml:**

```yaml
specVersion: "3.0"
metadata:
  name: testappforsap
type: application
framework:
  name: SAPUI5
  version: "1.84.0"
  libraries:
    - name: sap.ca.ui
    - name: sap.m
    - name: sap.ui.core
    - name: sap.ushell
    - name: themelib_sap_fiori_3
server:
  customMiddleware:
    - name: fiori-tools-proxy
      afterMiddleware: compression
      configuration:
        ignoreCertError: true
        backend:
          - path: /sap
            url: https://testforsap.com:44300
            ws: true
            client: "010"
        credentials:
          username: env:FIORI_TOOLS_USER
          password: env:FIORI_TOOLS_PASSWORD
    - name: fiori-tools-appreload
      afterMiddleware: compression
      configuration:
        port: 35729
        path: webapp
    - name: fiori-tools-servestatic
      afterMiddleware: compression
      configuration:
        paths:
          - path: /resources/sap|/test-resources/sap
            src: "/Users/Sam/Desktop/Development/sap/SAPUI5/custom-sapui5-library/src/sap"
builder:
  customTasks:
    - name: ui5-tooling-transpile-task
      afterTask: replaceVersion
      configuration:
        debug: true
        removeConsoleStatements: true
        transpileAsync: true
        excludePatterns:
          - "lib/"
          - "localService"
          - "test"
```

**package.json:**

```json
{
    "name": "testappforsap",
    "version": "0.0.1",
    "description": "",
    "private": true,
    "devDependencies": {
        "@openui5/ts-types": "^1.111.0",
        "@sap/ux-ui5-tooling": "^1.9.1",
        "@ui5/cli": "^3.0.3",
        "ui5-tooling-transpile": "^0.3.7"
    },
    "scripts": {
        "start": "ui5 serve -o test/flpSandbox.html#testapp-manage&/Orders",
        "build": "ui5 build preload --clean-dest --include-task=generateCachebusterInfo",
        "deploy": "npm run build && fiori deploy --config ui5-deploy.yaml -- -y && rimraf archive.zip"
    }
}
```

**Comment by flovogt** [comment]
Its indeed the reason I described above. I'll request an adjustment of the `ui5-tooling-transpile` package.

**Comment by flovogt** [comment]
This issue is not fixable by UI5 Tooling itself. Therefore closing this issue here. Further discussions can be added to the mentioned PR.

**Comment by flovogt** [comment]
Issue is fixed with `ui5-tooling-transpile` version `0.4.1`.
